src/xref2/ref_tools.ml

```diff
@@ -415,14 +406,30 @@ module CS = struct
     env_lookup_by_name Env.s_constructor name env
     >>= fun (`Constructor (id, _)) -> Ok (`Identifier id :> t)
 
-  let in_datatype _env ((parent', t) : datatype_lookup_result) name =
+  let got_a_field name =
+    (* Let's pretend we didn't see the field and say we didn't find anything. *)
+    Error (`Find_by_name (`Cons, name))
+
+  let in_parent _env (parent : label_parent_lookup_result) name =
```

Comment:
Perhaps we need a `_lookup_result` that correspond to `Reference.Parent` to remove the impossible cases ?